astronomer/providers/apache/livy/hooks/livy.py

```diff
@@ -0,0 +1,410 @@
+                    url,
+                    json=data if self.method in ("POST", "PATCH") else None,
+                    params=data if self.method == "GET" else None,
+                    headers=headers,
+                    auth=auth,
+                    **extra_options,
+                )
+                try:
+                    response.raise_for_status()
+                    return await response.json()
+                except ClientResponseError as e:
+                    self.log.warning(
+                        "[Try %d of %d] Request to %s failed.",
+                        attempt_num,
+                        self.retry_limit,
+                        url,
+                    )
+                    if not self._retryable_error_async(e) or attempt_num == self.retry_limit:
+                        self.log.error("HTTP error: %s", e)
+                        # In this case, the user probably made a mistake.
+                        # Don't retry.
+                        return {"Response": {e.message}, "Status Code": {e.status}, "status": "error"}
+
+                attempt_num += 1
+                await asyncio.sleep(self.retry_delay)
+
+    async def run_method(
+        self,
+        endpoint: str,
+        method: str = "GET",
+        data: Optional[Any] = None,
+        headers: Optional[Dict[str, Any]] = None,
+    ) -> Any:
+        """
+        Wrapper for HttpHookAsync, allows to change method on the same HttpHookAsync
+
+        :param method: http method
+        :param endpoint: endpoint
+        :param data: request payload
+        :param headers: headers
+        :return: http response
+        :rtype: requests.Response
+        """
+        if method not in ("GET", "POST", "PUT", "DELETE", "HEAD"):
+            return {"status": "error", "response": f"Invalid http method {method}"}
+
+        back_method = self.method
+        self.method = method
+        try:
+            result = await self._do_api_call_async(endpoint, data, headers, self.extra_options)
+        finally:
+            self.method = back_method
+        return {"status": "success", "response": result}
+
+    async def get_batch_state(self, session_id: Union[int, str]) -> Any:
+        """
+        Fetch the state of the specified batch asynchronously.
+
+        :param session_id: identifier of the batch sessions
+        :return: batch state
+        :rtype: BatchState
+        """
+        self._validate_session_id(session_id)
+        self.log.info("Fetching info for batch session %d", session_id)
+        result = await self.run_method(endpoint=f"/batches/{session_id}/state")
+        if result["status"] == "error":
+            self.log.info(result)
+            return {"batch_state": "error", "response": result, "status": "error"}
+
+        if "state" not in result["response"]:
+            self.log.info(
+                {"batch_state": "error", "response": f"Unable to get state for batch with id: {session_id}"}
+            )
```

Comment:
Dictionary in log line? Let's change it to string with params

Reply:
[Changed](https://github.com/astronomer/astronomer-providers/pull/178/commits/54da2a13e052ab18a5b1de906bda51ed14ac42bd)